File: Lab 6/C++/Container.cpp
#include "Container.h"
#include <iostream>
#include <algorithm>

void Container::addLine(const std::string& str) 
{
    texts.push_back(Text(str));// push_back з бібліотеки вектора 
}

void Container::removeLine(size_t index) 
{
    //texts.erase(texts.begin() + index);
    if (index < texts.size()) // перевірка чі індекс в межах 
    {
        texts.erase(texts.begin() + index);
    }
}

void Container::clearAll() 
{
    texts.clear();
}

const Text* Container::getLongestLine() const
{
    if (texts.empty())
    {
        return nullptr;
    }

    const Text* longest = &texts[0];
    for (size_t i = 0; i < texts.size(); ++i) 
    {
        const Text& t = texts[i];
        if (t.getLength() > longest->getLength()) 
        {
            longest = &t;
        }
    }

    return longest;
}

double Container::getNumberPercent() const 
{
    size_t digit_Count = 0;
    size_t total_Chars = 0;

    for (size_t i = 0; i < texts.size(); ++i)
    {
        digit_Count += texts[i].countNumbers();
        total_Chars += texts[i].getLength();
    }

    if (total_Chars == 0) 
    {
        return 0.0;
    }
    double percent = (double)digit_Count / total_Chars * 100;

    return percent;
}

size_t Container::getTotalLength() const 
{
    size_t total = 0; 

    for (size_t i = 0; i < texts.size(); ++i)
    {  
        total += texts[i].getLength(); 
    }

    return total; 
}

void Container::showAll() const
{
    for (size_t i = 0; i < texts.size(); ++i) 
    {
        std::cout << texts[i].getLine() << std::endl; 
    }
}

File: Lab 6/C#/Container.cs
using System;
using System.Collections.Generic;

public class Container
{
    private List<Text> texts = new List<Text>(); // Список об'єктів Text

    public void AddLine(string str)
    {
        texts.Add(new Text(str));
    }

    public void RemoveLine(int index)
    {
        if (index >= 0 && index < texts.Count)
        {
            texts.RemoveAt(index);
        }
    }

    public void ClearAll()
    {
        texts.Clear();
    }

    public Text GetLongestLine()
    {
        if (texts.Count == 0) return null;

        Text longest = texts[0];
        foreach (Text t in texts)
        {
            if (t.GetLength() > longest.GetLength())
            {
                longest = t;
            }
        }
        return longest;
    }

    public double GetNumberPercent()
    {
        int digitCount = 0;
        int totalChars = 0;

        foreach (Text t in texts)
        {
            digitCount += t.CountNumbers();
            totalChars += t.GetLength();
        }

        return totalChars == 0 ? 0.0 : (double)digitCount / totalChars * 100.0;
    }

    public int GetTotalLength()
    {
        int total = 0;
        foreach (Text t in texts)
        {
            total += t.GetLength();
        }
        return total;
    }

    public void ShowAll()
    {
        foreach (Text t in texts)
        {
            Console.WriteLine(t.GetLine());
        }
    }
}
